src/parse.rs

```diff
@@ -839,7 +867,7 @@ pub(crate) fn parse_id<'a>(s: &'a str, open: &'_ str, close: &'_ str) -> Option<
 }
 
 fn is_id_char(c: char) -> bool {
-    c.is_alphanumeric() || c == '_'
+    c.is_alphanumeric() || c == '_' || c == '-'
```

Comment:
Note that both Ruby and Python don't allow `-` in group names:

```
SyntaxError ((irb):14: invalid group name <foo-bar>>: /(?<foo-bar>\w+)%\k<foo-bar>/)
```

```
re.error: bad character in group name 'foo-bar' at position 4
```

But I now see that the reason you're doing this change is so that you can parse `\k<-1>`. The code needs to be changed around a bit. Can you add tests for these:

* `\k<foo-bar>` -> invalid group name
* `\k<-1>` -> works as a relative backref
* `\k<+1>` -> works as a relative backref (different from `\k<1>`)

To support that, I'd probably just check the first character within `parse_named_backref`. If it's a `-` or `+`, parse the rest as a number. Otherwise, parse the rest with `c.is_alphanumeric() || c == '_'`.